File: backend/app/models/competition.py
from sqlalchemy import Column, String, Text, DateTime, JSON
from sqlalchemy.orm import relationship
from app.core.database import Base


class Competition(Base):
    __tablename__ = "competitions"

    id = Column(String, primary_key=True, index=True)
    type = Column(String, nullable=False)  # race, puzzle, battle, fashion
    name = Column(String, nullable=False)
    description = Column(Text)
    start_date = Column(DateTime(timezone=True))
    end_date = Column(DateTime(timezone=True))
    rewards = Column(JSON)  # {first: [...], second: [...], third: [...]}
    status = Column(String, default="upcoming")  # upcoming, active, ended

    # 관계
    entries = relationship("CompetitionEntry", back_populates="competition")

File: backend-java/src/main/java/com/aetherisland/entity/Competition.java
package com.aetherisland.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "competitions")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Competition {
    @Id
    private String id;
    
    @Column(nullable = false)
    private String type; // race, puzzle, battle, fashion
    
    @Column(nullable = false)
    private String name;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(name = "start_date")
    private LocalDateTime startDate;
    
    @Column(name = "end_date")
    private LocalDateTime endDate;
    
    @Column(columnDefinition = "JSON")
    private String rewards; // JSON string: {first: [...], second: [...], third: [...]}
    
    @Column(nullable = false)
    private String status = "upcoming"; // upcoming, active, ended
    
    @OneToMany(mappedBy = "competition", cascade = CascadeType.ALL)
    private List<CompetitionEntry> entries;
}

